src/contexts/WalletContext/WalletContext.tsx

```diff
@@ -1,302 +1,98 @@
-      case "evm-wc":
-        disconnectEVMWC();
-        break;
-      case "binance-wallet":
-        disconnectBinanceWallet();
-        break;
-      case "xdefi-evm":
-        disconnectXdefi();
-        break;
-      case "keplr":
-        disconnectKeplr();
-        break;
-      case "keplr-wc":
-        disconnectKeplrWC();
-        break;
-      case "xdefi-wallet":
-      case "station":
-      case "leap-wallet":
-      case "walletconnect":
-        disconnectTerra();
-        break;
-      default:
-        throw new WalletDisconnectedError();
-    }
-  }, [
-    activeProvider?.providerInfo,
-    disconnectMetamask,
-    disconnectEVMWC,
-    disconnectBinanceWallet,
-    disconnectXdefi,
-    disconnectKeplr,
-    disconnectKeplrWC,
-    disconnectTerra,
-  ]);
-
-  const switchChain = useCallback(
-    async (chainId: chainIDs) => {
-      if (!activeProvider) {
-        throw new WalletDisconnectedError();
-      }
-
-      await activeProvider.switchChain(chainId);
-    },
-    [activeProvider]
+  const xdefiEvm = useInjectedProvider(
+    "xdefi-evm",
+    { logo: xdefiIcon, name: "Xdefi", supportedChains: EVMChains },
+    "https://chrome.google.com/webstore/detail/xdefi-wallet/hmeobnfnfcmdkdcmlblgagmfpfboieaf?hl=en"
   );
 
-  const {
-    data: chain = placeholderChain,
-    isLoading: isChainLoading,
-    isFetching: isChainFetching,
-    error,
-  } = useChainQuery(activeProvider?.providerInfo!, { skip: !activeProvider });
-
-  useVerifyChain(chain, error, disconnect);
-
-  const walletState: WalletState | undefined = useMemo(() => {
-    if (activeProvider) {
-      const { logo, providerId, address } = activeProvider.providerInfo!;
-      const walletState: WalletState = {
-        walletIcon: logo,
-        displayCoin: chain.native_currency,
-        coins: [chain.native_currency, ...chain.tokens],
-        address,
-        chain,
-        providerId,
-        supportedChains: activeProvider.supportedChains,
-      };
-
-      return walletState;
-    }
-  }, [activeProvider, chain]);
+  const { wc: stationMobile, extensions } = useTerra();
+  const keplr = useKeplr();
+  const keplrWC = useKeplrWC();
+  const evmWC = useEVMWC({
+    logo: metamaskIcon,
+    name: "Metamask Mobile",
+    supportedChains: EVMChains,
+  });
+
+  const wallets: Wallet[] = isMobile()
+    ? [evmWC, keplrWC, stationMobile]
+    : [
+        keplr,
+        metamask,
+        evmWC,
+        keplrWC,
+        binance,
+        xdefiEvm,
+        ...extensions,
+        stationMobile,
+      ];
+
+  const connectedWallet = wallets.find((w) => w.status === "connected") as
+    | ConnectedWallet
+    | undefined;
+
+  const isLoading = wallets.some((w) => w.status === "loading");
 
   return (
-    <getContext.Provider
-      value={{
-        wallet: walletState,
-        isLoading:
-          providerStatuses.some((x) => x.isLoading) ||
-          isChainLoading ||
-          isChainFetching,
-      }}
+    <context.Provider
```

Comment:
the simplification of wallet context interface is a result of improving individual wallet hook type definition